### PR TITLE
MGDOBR-1106: API versioning: Manager services and models

### PR DESCRIPTION
**Pull request by manstis**
[https://issues.redhat.com/browse/MGDOBR-1106](https://issues.redhat.com/browse/MGDOBR-1106)

No code was changed in this PR; only moved around.

All models/classes in `infra` or `manager` that relate to a particular _version_ have been moved into sub-packages.

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Comment by manstis** [comment]
It's probably more simple to fetch this locally and look at in your IDE..

There's no code changes; just a move around of `infra` and `manager`. There are other JIRA for `operator` and `processors`.

**Review by sutaakar — APPROVED** [review]

**Review by lucamolteni — APPROVED** [review]

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/APIConstants.java [hunk 1]
```diff
@@ -6,44 +6,49 @@
 public class APIConstants {
```
Comment (r00ta) [inline infra/src/main/java/com/redhat/service/smartevents/infra/api/APIConstants.java:6]: (side note) I think some constants in this class can be moved to "core" constants to be re-used in `v1` and `v2` services. On the other side I think it's good to do it in another PR so to keep this one scoped

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/BridgeResource.java [hunk 2]
```diff
@@ -5,9 +5,9 @@
 import com.redhat.service.smartevents.infra.api.APIConstants;
 import com.redhat.service.smartevents.integration.tests.common.BridgeUtils;
 import com.redhat.service.smartevents.integration.tests.common.Constants;
-import com.redhat.service.smartevents.manager.api.models.requests.BridgeRequest;
-import com.redhat.service.smartevents.manager.api.models.responses.BridgeListResponse;
-import com.redhat.service.smartevents.manager.api.models.responses.BridgeResponse;
+import com.redhat.service.smartevents.manager.api.v1.models.requests.BridgeRequest;
```
Comment (r00ta) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/BridgeResource.java:8]: you might have conflicts with https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1255 cc @sthundat

**Review by r00ta — APPROVED** [review]

**Review by r00ta — APPROVED** [review]

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/APIConstants.java (hunk 1 above)
Comment (manstis) [reply infra/src/main/java/com/redhat/service/smartevents/infra/api/APIConstants.java]: Done.

**Comment by manstis** [comment]
@sutaakar do you know if failure of the `Upgrade` test is known to be intermittent? The point of failure is "clean up" and should not be affected by this PR at all.

**Comment by sutaakar** [comment]
@manstis The last failure was caused by lack of resources for MC provisioning, now it should be addressed.
I have retriggered the test.

File: manager/src/main/java/com/redhat/service/smartevents/manager/workers/quartz/WorkJob.java [hunk 3]
```diff
@@ -41,6 +52,12 @@ private Worker<?> findWorker(Work work) {
             return bridgeWorker;
         } else if (Processor.class.getName().equals(work.getType())) {
             return processorWorker;
+        } else if (LEGACY_BRIDGE_WORKER_CLASSNAME.equals(work.getType())) {
+            // Existing Work may have been created on an older version
+            return legacyBridgeWorker;
```
Comment (robpblake) [inline manager/src/main/java/com/redhat/service/smartevents/manager/workers/quartz/WorkJob.java:57]: @manstis Do we need the extra `Legacy{Bridge|Processor}Worker` classes? IIUC we can avoid that with a simple switch statement?

```
switch(work.getType()) {
  case Bridge.class.getName():
  case LEGACY_BRIDGE_CLASSNAME:
      return bridgeWorker;
```

Or do I mis-understand the intent here?

File: manager/src/test/java/com/redhat/service/smartevents/manager/api/v1/internal/ShardBridgesSyncAPITest.java [hunk 4]
```diff
@@ -254,7 +264,57 @@ public void metricsAreProduced() {
...
 
-        metricsService.getMetricNames().forEach(metric -> assertThat(metrics).contains(metric));
+        // Not all metrics are recorded by this test. Only the "successful" path in this test.
+        assertThat(metrics).contains(operationTotalCountMetricName);
+        assertThat(metrics).contains(operationTotalSuccessCountMetricName);
+        assertThat(metrics).contains(operationDurationMetricName);
+    }
+
+    @Test
+    @TestSecurity(user = DEFAULT_CUSTOMER_ID)
```
Comment (robpblake) [inline manager/src/test/java/com/redhat/service/smartevents/manager/api/v1/internal/ShardBridgesSyncAPITest.java:274]: @manstis Is this change related?

**Review by robpblake — COMMENTED** [review]
@manstis Just a minor comment about the need for the `Legacy*Worker` classes. 

cheers,

Rob

File: manager/src/main/java/com/redhat/service/smartevents/manager/workers/quartz/WorkJob.java (hunk 3 above)
Comment (manstis) [reply manager/src/main/java/com/redhat/service/smartevents/manager/workers/quartz/WorkJob.java]: You're not missing anything. Refactored.

File: manager/src/test/java/com/redhat/service/smartevents/manager/api/v1/internal/ShardBridgesSyncAPITest.java (hunk 4 above)
Comment (manstis) [reply manager/src/test/java/com/redhat/service/smartevents/manager/api/v1/internal/ShardBridgesSyncAPITest.java:274]: @robpblake it is related; but for the life of me I cannot remember why. It has something to do with needing to split the "metrics" test into "success" and "failure" as not _ALL_ metrics were being (reliably) produced for the single test.

**Review by robpblake — APPROVED** [review]
